### PR TITLE
Aniorb Presence V2.0

### PR DESCRIPTION
**Pull request by ICEPrey**
My old PR request was closed because site had updates. So site updated and this is the new Aniorb presence.

### CODE REVIEW

**Comment by ICEPrey** [comment]
> My old PR request was closed because site had updates. So site updated and this is the new Aniorb presence.
Forgot about the images. 
watching







details
Movies
myList
popular
recent
Season

**Comment by ICEPrey** [comment]
Added genres 

genre

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Oh my, what happened with the formatting?

**Comment by ICEPrey** [comment]
> Oh my, what happened with the formatting?

wdym

**Comment by ririxidev** [comment]
> Oh my, what happened with the formatting?

it's okay? What are you talking about.

**Comment by EncryptedDev** [comment]
My bad, looked like the formatting was off, but I think it was because GitHub didn't render fully - my bad.

**Review by ICEPrey — COMMENTED** [review]
nothing to fix i guess?

File: websites/A/Aniorb/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "name": "ICEPrey",
+    "id": "256973318588989440"
+  },
+  "url": "aniorb.me",
+  "description": {
+    "en": "Aniorb is a free premium anime streaming website"
+  },
+  "service": "Aniorb",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/9qhIuod.png",
```
Comment (ririxidev) [inline websites/A/Aniorb/dist/metadata.json:13]: That's such a low quality logo.

File: websites/A/Aniorb/presence.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "aniorb",
+    startTimestamp: browsingStamp
+  };
+  if (document.location.pathname == "/") {
+    presenceData.details = "Browsing Aniorb";
+  } else if (document.location.pathname.includes("/recentlyadded")) {
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:44]:
```suggestion
  if (document.location.pathname == "/")
    presenceData.details = "Browsing Aniorb";
  else if (document.location.pathname.includes("/recentlyadded")) {
```

File: websites/A/Aniorb/presence.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    presenceData.details = "Browsing Recently Added";
+    presenceData.smallImageKey = "recent";
+    presenceData.buttons = [
+      {
+        label: "View Recently Added",
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:49]: This label doesn't make any sense.

File: websites/A/Aniorb/presence.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        label: "View Movies",
+        url: document.location.href
+      }
+    ];
+  } else if (document.location.pathname.includes("/popular")) {
+    presenceData.details = "Browsing Popular";
+    presenceData.smallImageKey = "star";
+    presenceData.buttons = [
+      {
+        label: "View Popular",
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:76]: This label doesn't make any sense.

File: websites/A/Aniorb/presence.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      presenceData.smallImageText = paused
+        ? (await strings).pause
+        : (await strings).play;
+      presenceData.endTimestamp = timestamps[1];
+      if (paused) {
+        delete presenceData.endTimestamp;
+      }
+    }
+  } else if (document.location.pathname.includes("/search")) {
+    presenceData.details = "Searching For...";
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:167]: I don't think you should include ``...`` when you are setting search text in state.

File: websites/A/Aniorb/presence.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        label: "View Recently Added",
+        url: document.location.href
+      }
+    ];
+  } else if (document.location.pathname.includes("/newseason")) {
+    presenceData.details = "Browsing New Season";
+    presenceData.smallImageKey = "season";
+    presenceData.buttons = [
+      {
+        label: "View New Season",
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:58]: this label doesn't make any sense.

File: websites/A/Aniorb/presence.ts [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        label:
+          "View The Genre " +
+          document.querySelector(
+            "span.text-red-500.capitalize.px-16.font-thin.text-xl"
+          ).textContent,
+        url: document.location.href
+      }
+    ];
+  } else if (document.location.pathname.includes("/details")) {
+    presenceData.details = `Checking Synopsis`;
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:98]: ` -> "

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/A/Aniorb/presence.ts (hunk 3 above)
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:49]: Wdym it doesn't make any sense? Can you elaborate please?

Comment (ririxidev) [reply websites/A/Aniorb/presence.ts:49]: View recently added what... Site? Subpage?

Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:49]: Takes you to the website's Recently Added page?

**Comment by ICEPrey** [comment]
Sub page

File: websites/A/Aniorb/presence.ts (hunk 3 above)
Comment (ICEPrey) [reply websites/A/Aniorb/presence.ts:49]: sub page

Comment (ririxidev) [reply websites/A/Aniorb/presence.ts:49]: Okay but this label still doesn't make any sense with only these 3 words.

File: websites/A/Aniorb/dist/metadata.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    "id": "256973318588989440"
+  },
+  "url": "aniorb.me",
+  "description": {
+    "en": "Aniorb is a free premium anime streaming website"
+  },
+  "service": "Aniorb",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/U53Lfa5.png",
+  "thumbnail": "https://i.imgur.com/sFTMLrc.png",
```
Comment (Bas950) [inline websites/A/Aniorb/dist/metadata.json:14]: logo must be at least 512x512

File: websites/A/Aniorb/presence.ts [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+        label: "View Popular Animes",
+        url: document.location.href
+      }
+    ];
+  } else if (document.location.pathname.includes("/genre")) {
+    presenceData.details =
+      "Viewing The Genre " +
+      document.querySelector(
+        "span.text-red-500.capitalize.px-16.font-thin.text-xl"
+      ).textContent;
```
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:85]:
```suggestion
    presenceData.details =
      "Viewing genre:";
    presenceData.state = 
      document.querySelector(
        "span.text-red-500.capitalize.px-16.font-thin.text-xl"
      ).textContent;
```

File: websites/A/Aniorb/presence.ts [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    presenceData.buttons = [
+      {
+        label:
+          "View The Genre " +
+          document.querySelector(
+            "span.text-red-500.capitalize.px-16.font-thin.text-xl"
+          ).textContent,
+        url: document.location.href
+      }
+    ];
```
Comment (Bas950) [inline websites/A/Aniorb/presence.ts:96]:
```suggestion
    presenceData.buttons = [
      {
        label: "View Genre ",
        url: document.location.href
      }
    ];
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by ririxidev — DISMISSED** [review]

File: websites/A/Aniorb/presence.ts [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+    } else {
+      video = document.querySelector("div > div.plyr__video-wrapper > video");
+      if (video) {
+        (currentTime = video.currentTime),
+          (duration = video.duration),
+          (paused = video.paused),
+          (timestamps = presence.getTimestamps(
+            Math.floor(currentTime),
+            Math.floor(duration)
+          ));
```
Comment (ririxidev) [inline websites/A/Aniorb/presence.ts:147]: aaa wait, you don't need () there.

Comment (Bas950) [inline websites/A/Aniorb/presence.ts:147]:
```suggestion
        currentTime = video.currentTime;
        duration = video.duration;
        paused = video.paused;
        timestamps = presence.getTimestamps(
            Math.floor(currentTime),
            Math.floor(duration)
          ));
```

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**